client/src/main/java/io/pravega/client/stream/impl/ControllerResolverFactory.java

```diff
@@ -121,10 +143,10 @@ public String getServiceAuthority() {
         @Override
         @Synchronized
         public void start(Listener listener) {
```

Comment:
Because it deals with threads for lifecycle methods correctly. As it stands the code does not. It either needs to add locking everywhere the executor is refrenced or make it an atomic reference.